frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPartition.jsx

```diff
@@ -30,104 +31,90 @@ const DragWrapper = styled.div`
   }
 `;
 
-// eslint-disable-next-line no-unused-vars
-class ShowTotalsOption extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { showTotals: true };
-  }
-  toggleTotals = () => {
-    const { showTotals } = this.state;
-    this.setState({ showTotals: !showTotals });
-    this.props.onChangeSettings(!showTotals);
-  };
-
-  render() {
-    const { showTotals } = this.state;
-    return (
-      <Flex pt={2} justifyContent="space-between" alignItems="center">
-        <Text>{t`Show totals`}</Text>
-        <Toggle value={showTotals} onChange={this.toggleTotals}></Toggle>
-      </Flex>
-    );
+function ShowTotalsOption({ value, onChange }) {
+  if (value === null) {
+    return null;
   }
+  return (
+    <Flex pt={2} justifyContent="space-between" alignItems="center">
+      <Text>{t`Show totals`}</Text>
+      <Toggle value={value} onChange={() => onChange(!value)}></Toggle>
+    </Flex>
+  );
 }
 
-class SortButton extends React.Component {
-  render() {
-    const { iconName, onChange, currentValue, buttonValue } = this.props;
-    const isSelected = buttonValue === currentValue;
-    return (
-      <Icon
-        name={iconName}
-        onClick={() => onChange(isSelected ? undefined : buttonValue)}
-        size={16}
-        className={cx("sort cursor-pointer", {
-          "text-brand": isSelected,
-          "text-medium text-brand-hover": !isSelected,
-        })}
-      />
-    );
-  }
+function SortButton({ iconName, onChange, currentValue, buttonValue }) {
```

Comment:
The class->function switches here are unrelated to the change. The component above no longer needed state, so I switched them all over to match.